src/library/scala/collection/immutable/Queue.scala

```diff
@@ -100,6 +100,15 @@ class Queue[+A] protected(protected val in: List[A], protected val out: List[A])
     case _                               => super.:+(elem)(bf)
   }
 
+  override def ++[B >: A, That](that: GenTraversableOnce[B])(implicit bf: CanBuildFrom[Queue[A], B, That]): That = {
+    if (bf eq Queue.ReusableCBF) {
+      val thatQueue = that.asInstanceOf[Queue[B]]
+      new Queue[B](thatQueue.in ++ (thatQueue.out.reverse ++ this.in), this.out).asInstanceOf[That]
```

Comment:
`thatQueue.out.reverse ++ this.in` copies `thatQueue.out` twice — I think you can get it down to one copy by using `List#reverse_:::`, a trick you'll see used in the body of `def enqueue[B >: A](iter: Iterable[B])`